SupportYourLocals.Data/XMLData.cs

```diff
@@ -0,0 +1,142 @@
+
+                    var productsList = new List<string>();
+                    foreach (XElement product in productNode)
+                    {
+                        productsList.Add(product.Value);
+                    }
+                    dictionary.Add(productTypeEnum, productsList);
+                }
+                localSellersDictionary.Add(id, new LocationData(location, name, addedById, time, dictionary));
+                
+            }
+            return localSellersDictionary;
+        }
+
+        public void SaveData()
+        {
+            doc = new XDocument(new XElement("LocalSellers"));
+            foreach (var data in dictionaryLocationDataById.Values)
+            {
+                XElement root = new XElement("LocalSeller");
+                root.Add(new XAttribute("ID", data.ID));
+                root.Add(new XAttribute("Location", data.Location));
+                root.Add(new XAttribute("Name", data.Name));
+                root.Add(new XAttribute("AddedByID", data.AddedByID));
+                root.Add(new XAttribute("Time", data.Time));
+                AddProductTypesToXml(data, root);
+                doc.Element("LocalSellers").Add(root);
+                doc.Save(filePath);
+            }
+        }
+
+        private void AddProductTypesToXml(LocationData data, XElement root)
+        {
+            foreach (var productType in data.Products)
+            {
+                if (productType.Value[0] != "")
+                {
+                    XElement productTypeBranch = new XElement("ProductType");
+                    productTypeBranch.Add(new XAttribute("type", productType.Key));
+                    foreach (var product in productType.Value)
+                    {
+                        if (product != "")
+                        {
+                            XElement productBranch = new XElement("Product");
+                            productBranch.Add(content: product);
+                            productTypeBranch.Add(productBranch);
+                        }
+                    }
+                    root.Add(productTypeBranch);
+                }
+            }
+        }
+        public void AddData(LocationData data)
+        {
+            dictionaryLocationDataById.Add(data.ID, data);
+        }
+
+        public List<LocationData> GetAllData()
+        {
+            return dictionaryLocationDataById.Select(d => d.Value).ToList();
+        }
+
+        public LocationData GetData(string id)
+        {
+            return dictionaryLocationDataById[id];
+        }
+
+        public int GetDataCount()
+        {
+            return dictionaryLocationDataById.Count;
+        }
+
+        public void RemoveData(string id)
+        {
+            dictionaryLocationDataById.Remove(id);
+        }
+
+        public void UpdateData(LocationData data)
+        {
+            RemoveData(data.ID);
+            AddData(data);
```

Comment:
Can be simplified a bit.

```suggestion
            dictionaryLocationDataById[data.ID] = data;
```